### ISSUE TITLE
[BUG] Note information note showing up

### ISSUE THREAD

**Issue by RenegadeButcher** [getAlby/lightning-browser-extension#2312]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

If I generate an invoice with my Alby wallet, regardless of if it's with the browser extenion, linked to Blue Wallet or through the Wordpress extension, notes added during payments do not seem to appear.  I've tested this both with Blue Wallet and Wallet of Satoshi on the sending side.  Note information isn't appearing in the wallet, or logged in to the Alby website.

Having reliable notes come through would be a huge boon.  I'd like to enable 1-time lightning sales for products, but without a user being able to attach contact or shipping info it's a mess and requires a separate conversation with the customer through other means.

### Steps To Reproduce

1. Generate an invoice to an Alby Wallet
2. Pay said invoice with Wallet of Satoshi
3. Include some data in the payment note
4. Observe the lack of information on the receiving side

### Expected behavior

Payment note information should be available and usable across all lightning wallets

### Alby information

Current plugin, website and LNDhub linked wallets

### Are you working on this?

None

**Comment by reneaaron** [comment]
Thanks for your report. 

To my knowledge there is no way to attach a comment to an invoice. (i see that WoS allows to add that in their app, but that won't work on a protocol level, also I know no other app that allows that, also checked BlueWallet & ZEUS) 

You can only attach comments as you receive to a lightning address (or any LNURL). This is also why it works with the widget (there you send to an LNURL and the comment is added in the UI of the widget, if you pay the generated QR code with WoS the comment will show up on both your Alby account and the extension). 

I think the problem here really is WoS displaying the "add note" for invoices (while it should only be in there for LNURL payments).

**Comment by RenegadeButcher** [comment]
Hmmmmm, interesting.  So, if I send a Zap on Nostr from my Alby wallet, I
can attach a note and it will carry through.  Even if I do so on something
like Snort, and pay the generated invouce with my Alby wallet.  Now, that
comment isn't entered on the wallet, but as I setup the Zap.  Perhaps
that's the difference?  It's possible to only attach the note at the time
the invoice is generated?

**Comment by reneaaron** [comment]
Zaps work a bit  different on a technical level, it's different from Lightning Address payments. If you pay to your getAlby Lightning Address from WoS the comment will also show up.

**Comment by RenegadeButcher** [comment]
Hmmm, interesting.  Would there be a way to generate a payment of a set
amount to a lightning address with a button VS creating an invoice?

**Comment by RenegadeButcher** [comment]
Or here's another thought...could the pay wall potentially open a new
windows or pop-up?  I could use a paywall widget as a payment button then,
that could open a page with a contact form where the customer could enter
their information.  I thought about using the pay wall and placing a form
below, but it seems a bit clunky and having a min 1hr could be
problematic.  Maybe I can come up with a form that is one time use 🤔

**Comment by reneaaron** [comment]
> Would there be a way to generate a payment of a set amount to a lightning address with a button VS creating an invoice?

Yes, that should be possible. However, that would need a specific lightning address that only allows payments for the given amount. Did you consider using BTCPay Server for that? I think they also allow customers to add information to an order.

**Comment by RenegadeButcher** [comment]
I have thought about BTCpay or LNbits, but I don't currently have the data
connection to run a node for hot payments.  Hopefully soon.

**Comment by RenegadeButcher** [comment]
I think there are definitely way it could work, but it's probably down to
me misunderstanding some intricacies of the protocol.  I try 😂

I have success generating invoices for single purchases, but was hoping
there was a way to implement this in something much like the donation
button widget.  Then an invoice is generated for the user on demand, it
just doesn't work for sales if I don't know who paid lol

And if I generate an invoice, it works one time.  This would hopefully be a
single recurring button per item.

Yes, I can use BTCpay server or LNbits to accomplish this.  But I was
thinking this would be a really slick way to get the average WordPress user
into simple lightning sales without needing to run a node or buy space on
someone else's.  WordPress is so widespread, a really simple solution like
this would make mass adoption much easier.

And maybe it's not possible with the current way the protocol works?
